src/coreclr/jit/hwintrinsicxarch.cpp

```diff
@@ -2027,9 +2029,12 @@ GenTree* Compiler::impSpecialIntrinsic(NamedIntrinsic        intrinsic,
             if (IsBaselineVector512IsaSupportedOpportunistically())
             {
                 op1 = impSIMDPopStack();
-                op1 =
-                    gtNewSimdHWIntrinsicNode(TYP_MASK, op1, NI_AVX512F_ConvertVectorToMask, simdBaseJitType, simdSize);
 
+                if (!varTypeIsMask(op1))
```

Comment:
hhm, where we previously creating the node `NI_AVX512F_ConvertVectorToMask` even for `op1 == TYP_MASK`?

Reply:
I didn't hit a case, but it's a place I wanted to be more explicit around and ensure that we weren't making a bad assumption (especially as the general logic for inserting the conversions is getting changed for both x64 and Arm64 to improve general handling).